src/extension.ts

```diff
@@ -41,60 +48,101 @@ export async function activate(context: ExtensionContext): Promise<void> {
 				command: `jsonnet.evalItem`,
 				arguments: [evalFilePath(editor), editor.selection.active],
 			};
-			evalAndDisplay(params, false);
+			evalAndDisplay(params, OutputMode.json);
 		}),
-		commands.registerCommand('jsonnet.evalFile', evalFileFunc(false)),
-		commands.registerCommand('jsonnet.evalFileYaml', evalFileFunc(true)),
-		commands.registerCommand('jsonnet.evalExpression', evalExpressionFunc(false)),
-		commands.registerCommand('jsonnet.evalExpressionYaml', evalExpressionFunc(true))
+		commands.registerCommand('jsonnet.evalFile', evalFileFunc(OutputMode.json)),
+		commands.registerCommand('jsonnet.evalFileYaml', evalFileFunc(OutputMode.yaml)),
+		commands.registerCommand('jsonnet.evalFileString', evalFileFunc(OutputMode.string)),
+		commands.registerCommand('jsonnet.evalExpression', evalExpressionFunc(OutputMode.json)),
+		commands.registerCommand('jsonnet.evalExpressionYaml', evalExpressionFunc(OutputMode.yaml)),
+		commands.registerCommand('jsonnet.evalExpressionString', evalExpressionFunc(OutputMode.string))
 	);
 }
 
-function evalFileFunc(yaml: boolean) {
+/**
+ * @returns The extension of the output file suggested by the header of the current file.
+ */
+function suggestOutputExtension(): string {
+    const editor = window.activeTextEditor;
+    const header = editor.document.lineAt(0).text;
+    const ext = /output: \S*[.](?<ext>\S*)/i.exec(header)?.groups?.ext;

```

Comment:
It does work but the `Output: ...` remains in the evaluated file which is a bit weird. Could it instead be a dialog option? (like the expression)